src/app/pages/datasets/components/roles-card/roles-card.component.ts

```diff
@@ -48,4 +55,37 @@ export class RolesCardComponent {
     }
     return shareNamesPretty;
   }
+
+  get canCreateShare(): boolean {
+    return !this.hasChildrenWithShares
+      && !this.isSystemDataset
+      && !this.isApplications
+      && !this.dataset.apps?.length
+      && !this.dataset.vms?.length
+      && !this.dataset.smb_shares?.length
+      && !this.dataset.nfs_shares?.length
+      && !this.dataset.iscsi_shares?.length;
+  }
+
+  constructor(
+    private slideInService: IxSlideInService,
+    private datasetStore: DatasetTreeStore,
+  ) {}
+
+  createSmbShare(): void {
+    const slideInRef = this.slideInService.open(SmbFormComponent);
+    slideInRef.componentInstance.form.patchValue({ path: this.dataset.mountpoint });
+    slideInRef.componentInstance.setNameFromPath();
+    slideInRef.slideInClosed$.pipe(filter(Boolean), untilDestroyed(this)).subscribe(() => {
+      this.datasetStore.datasetUpdated();
+    });
+  }
+
+  createNfsShare(): void {
+    const slideInRef = this.slideInService.open(NfsFormComponent);
+    slideInRef.componentInstance.form.patchValue({ path: this.dataset.mountpoint });
```

Comment:
Form is a private property in spirit. It would be better to pass in additional params via `data` in constructor.